waiter/src/waiter/discovery.clj

```diff
@@ -89,14 +104,39 @@
     (zipmap (map #(str (.getId %)) filtered-instances)
             (map #(get-instance-url % protocol endpoint) filtered-instances))))
 
+(defn- router-details
+  "Returns a map representing the details of the given router."
+  [instance]
+  (let [custom-details (pc/map-keys keyword (.getPayload instance))
+        details {:address (.getAddress instance)
+                 :custom-details custom-details
+                 :id (.getId instance)
+                 :name (.getName instance)
+                 :port (.getPort instance)}]
+    details))
+
+(defn router-id->details
```

Comment:
Nitpick: rename fucntion:
```suggestion
(defn retrieve-router-id->details
```

Reply:
I'm happy to rename - I just followed the pattern established by `router-id->endpoint-url` above.